### DISCUSSION TITLE
Reply from Submission screen

### DISCUSSION THREAD

**Discussion by shifuma** [verbb/formie#1920] [General]
I'm exploring an idea to reply to directly to Formie submissions from the submissions page. Do you have any tips or perhaps things to be wary of if I did go down this path?

From a bit of a look around, [it seems like](https://verbb.io/craft-plugins/formie/docs/developers/hooks#control-panel-edit-submission) `Craft::$app->getView()->hook('formie.cp.submissions.edit.content', function(array &$context) {}` might be the way to do it, and then handle the rest of the logic in a module. Would that be right?

**Comment by engram-design** [comment]
I'm not sure what functionality you're planning to incorporate into "replying", but that'd be the first step yep! Similar to how you can add extra Twig/JS into Entries and other elements, you can get Formie to render content using that event.

Everything else should be handled in your own module with your own logic.

**Reply by shifuma** [reply]
Thanks! Sorry it wasn't clear, I just mean to reply directly to the person who submitted the form. I'll look into it some more.

**Reply by engram-design** [reply]
Gotcha! I would say we'd introduce something like this to Formie, but it's also heavily reliant on having an email field in the form, so it's not going to be for everyone. But yes, entirely within the realm of module development.

**Reply by shifuma** [reply]
Would be a cool feature but understand not for everyone. The other trouble I'm running into is how to receive new replies back into the same submission. Not sure if that would even be possible.

**Reply by engram-design** [reply]
I suppose it depends on how far you want to take this concept - what if there starts to be an email thread for this, with a few replies back-and-forth? That starts becoming more like a support ticket sort of thing.

But yeah, I imagine you want to record the replies against the submission, but the replies are recorded separately as records of their own. In a similar fashion to something like the Comments plugin might do, where you can "comment" on a Formie submission element. That'd be a reply recorded against the submission. But I don't know how you plan to facilitate a reply - is that another Formie Submission, or an email reply? If the latter, that's more complicated (but could be done).

**Reply by shifuma** [reply]
Yep, probably more complicated than it needs to be, and better in a dedicated system. Thanks for your thoughts!
